core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt

```diff
@@ -0,0 +1,1700 @@
+import org.kiwix.kiwixmobile.core.dao.NewBookmarksDao
+import org.kiwix.kiwixmobile.core.downloader.fetch.DOWNLOAD_NOTIFICATION_TITLE
+import org.kiwix.kiwixmobile.core.extensions.ViewGroupExtensions.findFirstTextView
+import org.kiwix.kiwixmobile.core.extensions.snack
+import org.kiwix.kiwixmobile.core.extensions.toast
+import org.kiwix.kiwixmobile.core.main.DocumentParser.SectionsListener
+import org.kiwix.kiwixmobile.core.main.KiwixTextToSpeech.OnInitSucceedListener
+import org.kiwix.kiwixmobile.core.main.KiwixTextToSpeech.OnSpeakingListener
+import org.kiwix.kiwixmobile.core.main.MainMenu.MenuClickListener
+import org.kiwix.kiwixmobile.core.main.TableDrawerAdapter.DocumentSection
+import org.kiwix.kiwixmobile.core.main.TableDrawerAdapter.TableClickListener
+import org.kiwix.kiwixmobile.core.page.bookmark.adapter.BookmarkItem
+import org.kiwix.kiwixmobile.core.page.history.adapter.HistoryListItem.HistoryItem
+import org.kiwix.kiwixmobile.core.reader.ZimFileReader
+import org.kiwix.kiwixmobile.core.reader.ZimReaderContainer
+import org.kiwix.kiwixmobile.core.utils.AnimationUtils.rotate
+import org.kiwix.kiwixmobile.core.utils.ExternalLinkOpener
+import org.kiwix.kiwixmobile.core.utils.LanguageUtils
+import org.kiwix.kiwixmobile.core.utils.LanguageUtils.Companion.getCurrentLocale
+import org.kiwix.kiwixmobile.core.utils.LanguageUtils.Companion.handleLocaleChange
+import org.kiwix.kiwixmobile.core.utils.REQUEST_STORAGE_PERMISSION
+import org.kiwix.kiwixmobile.core.utils.REQUEST_WRITE_STORAGE_PERMISSION_ADD_NOTE
+import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil
+import org.kiwix.kiwixmobile.core.utils.StyleUtils.getAttributes
+import org.kiwix.kiwixmobile.core.utils.TAG_CURRENT_ARTICLES
+import org.kiwix.kiwixmobile.core.utils.TAG_CURRENT_FILE
+import org.kiwix.kiwixmobile.core.utils.TAG_CURRENT_POSITIONS
+import org.kiwix.kiwixmobile.core.utils.TAG_CURRENT_TAB
+import org.kiwix.kiwixmobile.core.utils.TAG_FILE_SEARCHED
+import org.kiwix.kiwixmobile.core.utils.TAG_FILE_SEARCHED_NEW_TAB
+import org.kiwix.kiwixmobile.core.utils.TAG_KIWIX
+import org.kiwix.kiwixmobile.core.utils.UpdateUtils.reformatProviderUrl
+import org.kiwix.kiwixmobile.core.utils.dialog.DialogShower
+import org.kiwix.kiwixmobile.core.utils.dialog.KiwixDialog
+import org.kiwix.kiwixmobile.core.utils.files.FileUtils.deleteCachedFiles
+import org.kiwix.kiwixmobile.core.utils.files.FileUtils.readFile
+import java.io.File
+import java.io.IOException
+import java.text.SimpleDateFormat
+import java.util.Date
+import javax.inject.Inject
+import kotlin.math.abs
+import kotlin.math.max
+
+@Suppress("LargeClass")
+abstract class CoreReaderFragment :
+  BaseFragment(),
+  WebViewCallback,
+  MenuClickListener,
+  FragmentActivityExtensions,
+  WebViewProvider {
+  protected val webViewList: MutableList<KiwixWebView> = ArrayList()
+  private val webUrlsProcessor = BehaviorProcessor.create<String>()
+
+  @JvmField @BindView(R2.id.toolbar)
+  var toolbar: Toolbar? = null
```

Comment:
Can we have two lines for project consistency?

Reply:
hi @gouri-panda  Done.